components/selection/selection-header.js

```diff
@@ -127,10 +128,10 @@ export class SelectionHeader extends SelectionObserverMixin(RtlMixin(LocalizeCor
 
 	render() {
 		const classes = this._getSelectionHeaderContainerClasses();
-		const label = this._getSelectionHeaderLabel();
+		const label = (this._hasActions || !this.noSelection) ? this._getSelectionHeaderLabel() : null;
```

Comment:
I think this will have to move into `this._getSelectionHeaderLabel` for the table header to work, because its conditions will be a bit different (since it doesn't display a select-all checkbox). But I can make that change in my PR.

Reply:
Ah ok, sounds good! Yeah I wanted to keep it out of `list-header` if it was just going to be the exact same in both places, but once table is different feel free to move it as needed.  Once that happens it probably makes sense to also add tests for it in `list` and `table` accordingly